src/core/thread/network_data_service.cpp

```diff
@@ -100,29 +100,36 @@ void Manager::GetBackboneRouterPrimary(ot::BackboneRouter::BackboneRouterConfig
 {
     const ServerTlv *                 rvalServerTlv  = nullptr;
     const BackboneRouter::ServerData *rvalServerData = nullptr;
-    Iterator                          iterator;
 
     aConfig.mServer16 = Mac::kShortAddrInvalid;
 
-    iterator.mServiceTlv = Get<Leader>().FindService(kThreadEnterpriseNumber, &BackboneRouter::kServiceData,
-                                                     sizeof(BackboneRouter::kServiceData));
-
-    while (IterateToNextServer(iterator) == kErrorNone)
+    for (const ServiceTlv *serviceTlv = Get<Leader>().FindService(
+             kThreadEnterpriseNumber, &BackboneRouter::kServiceData, 1, NetworkData::kServiceDataPrefixMatch);
```

Comment:
define a constant, like `THREAD_SERVICE_DATA_BBR_SIZE` as 1, in the header and avoid the magic value 1 here?

Reply:
FYI Below in the code, instead of '1' ,  the expression 'sizeof(serviceData)' was used. Whether this or constant or numeral 1, it's best to keep a consistent approach.

Reply:
Refer to https://github.com/openthread/openthread/pull/6752#discussion_r662985288 for concern on `sizeof` approach

Reply:
may use `sizeof(BackboneRouter::kServiceData)` to avoid magic number 1 for consistency.

Reply:
Using sizeof() for a constant that should remain 1 also in the future, seems risky (even when it is 'protected' by unit tests), because size of BackboneRouter::kServiceData may increase one day while the constant 1 here is not meant to go along with that increase. In case a constant is used it should be like "THREAD_SERVICE_DATA_BBR_MIN_SIZE" or so because it's not about the current size of the data structure but about its minimum size.  In any case, I'm okay with any resolution of this including keeping the magic number :)

Reply:
Agree with @EskoDijk